### PR TITLE
lbmap: Sort backends before creating maglev lookup table

### LINKED ISSUE #13224: CI: K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with direct routing Tests NodePort with Maglev Tests Maglev backend selection
## CI failure

```
/home/jenkins/workspace/Cilium-PR-K8s-1.12-net-next/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Expected
    <string>: echo-5dc44bffc6-txwsd
to equal
    <string>: echo-5dc44bffc6-nsst5
/home/jenkins/workspace/Cilium-PR-K8s-1.12-net-next/src/github.com/cilium/cilium/test/k8sT/Services.go:1658

```
https://jenkins.cilium.io/job/Cilium-PR-K8s-1.12-net-next/519/testReport/junit/Suite-k8s-1/12/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_Tests_with_direct_routing_Tests_NodePort_with_Maglev_Tests_Maglev_backend_selection/

[22314fb4_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_Tests_with_direct_routing_Tests_NodePort_with_Maglev_Tests_Maglev_backend_selection.zip](https://github.com/cilium/cilium/files/5250500/22314fb4_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_Tests_with_direct_routing_Tests_NodePort_with_Maglev_Tests_Maglev_backend_selection.zip)

cc @borkmann  seems important

### PR DESCRIPTION
**Pull request by brb**
Maglev algorithm might produce different lookup table for the same set of backends when they are listed in a different order. To avoid that sort backends by name, because the names are the same on all nodes, while IDs are not.

Fix: 51605b63 ("Add pkg/maglev")
Fixes: https://github.com/cilium/cilium/issues/13224

### CODE REVIEW

**Comment by brb** [comment]
retest-net-next

**Review by pchaigno — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Comment by brb** [comment]
retest-net-next

File: pkg/maps/lbmap/lbmap.go.orig [hunk 1]
```diff
@@ -0,0 +1,606 @@
+// Copyright 2016-2019 Authors of Cilium
```
Comment (pchaigno) [inline pkg/maps/lbmap/lbmap.go.orig]: Looks like this was maybe committed by mistake?

Comment (brb) [reply pkg/maps/lbmap/lbmap.go.orig]: :facepalm:

**Comment by brb** [comment]
retest-net-next

**Comment by brb** [comment]
retest-4.19

**Comment by brb** [comment]
retest-runtime

**Review by aanm — APPROVED** [review]

**Pull request merged**